### ISSUE TITLE
Compilation failure related to WebSocket.send and ArrayBufferLike

### ISSUE THREAD

**Issue by BTOdell** [Microsoft/TypeScript#22916]
**TypeScript Versions:**
typescript@2.8.1
typescript@2.9.0-dev.20180327

**Search Terms:** 
WebSocket.send
SharedArrayBuffer
ArrayBufferLike

**Code**

Test code:
```ts
declare function createArrayBuffer(): ArrayBufferLike;
new WebSocket("").send(createArrayBuffer());
```

The typings for WebSocket.send are in lib.es6.d.ts:
```diff
- send(data: any): void;
+ send(data: USVString | ArrayBuffer | Blob | ArrayBufferView): void;
```

The typings for ArrayBufferLike are in lib.es6.d.ts as well:
```ts
interface ArrayBufferTypes {
    ArrayBuffer: ArrayBuffer;
}
type ArrayBufferLike = ArrayBufferTypes[keyof ArrayBufferTypes];
```

**Update**: lib.es2017.sharedmemory.d.ts defines:
```ts
interface ArrayBufferTypes {
    SharedArrayBuffer: SharedArrayBuffer;
}
```

~~I don't know why the compiler is reporting this as an error when the WebSocket class accepts the 'any' type as the parameter for its send function.~~
~~ArrayBufferLike doesn't even include SharedArrayBuffer.~~

**Expected behavior:**
For the code to compile successfully.

**Actual behavior:**
Compilation fails with:
```
src/ts/test.ts(2,24): error TS2345: Argument of type 'ArrayBuffer | SharedArrayBuffer' is not assignable to parameter of type 'string | ArrayBuffer | Blob | ArrayBufferView'.
  Type 'SharedArrayBuffer' is not assignable to type 'string | ArrayBuffer | Blob | ArrayBufferView'.
    Type 'SharedArrayBuffer' is not assignable to type 'ArrayBufferView'.
```

**Playground Link:**
N/A: Playground is not running up-to-date TypeScript version.

**Additional Information:**
This code compiles successfully in TypeScript 2.7.1.
I ran this in a brand new project, so there is no chance other packages or typings are conflicting.

**Comment by mhegazy** [comment]
A PR would be appreciated.

**Comment by mhegazy** [comment]
PRs welcomed. You can find more information about contributing lib.d.ts fixes at https://github.com/Microsoft/TypeScript/blob/master/CONTRIBUTING.md#contributing-libdts-fixes.

**Comment by BTOdell** [comment]
It seems like the compilation failure is due to a recent change to the WebSocket interface in lib.es6.d.ts:
```diff
- send(data: any): void;
+ send(data: USVString | ArrayBuffer | Blob | ArrayBufferView): void;
```
I'll work on a pull request to change it to:
```diff
- send(data: USVString | ArrayBuffer | Blob | ArrayBufferView): void;
+ send(data: USVString | ArrayBufferLike | Blob | ArrayBufferView): void;
```
